internal/webhook/dscinitialization/validating.go

```diff
@@ -0,0 +1,93 @@
+//go:build !nowebhook
+
+package dscinitialization
+
+import (
+	"context"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/webhook/shared"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+)
+
+//+kubebuilder:webhook:path=/validate-dscinitialization,mutating=false,failurePolicy=fail,sideEffects=None,groups=dscinitialization.opendatahub.io,resources=dscinitializations,verbs=create;delete,versions=v1,name=dscinitialization-validator.opendatahub.io,admissionReviewVersions=v1
+//nolint:lll
+
+// Validator implements webhook.AdmissionHandler for DSCInitialization validation webhooks.
+// It enforces singleton creation and deletion rules for DSCInitialization resources.
+type Validator struct {
+	Client  client.Client
+	Decoder admission.Decoder
+	Name    string
+}
+
+// Assert that Validator implements admission.Handler interface.
+var _ admission.Handler = &Validator{}
+
+// InjectDecoder implements admission.DecoderInjector so the manager can inject the decoder automatically.
+func (v *Validator) InjectDecoder(d admission.Decoder) error {
+	v.Decoder = d
+	return nil
+}
+
+// SetupWithManager registers the validating webhook with the provided controller-runtime manager.
+// It configures the webhook to handle DSCInitialization resources and uses a custom log constructor for context.
+func (v *Validator) SetupWithManager(mgr ctrl.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register("/validate-dscinitialization", &webhook.Admission{
+		Handler:        v,
+		LogConstructor: shared.NewLogConstructor(v.Name),
+	})
+	// No error to return currently, but return nil for future extensibility
+	return nil
+}
+
+// validateDupCreation denies creation if another instance of DSCInitialization already exists (singleton enforcement).
+func (v *Validator) validateDupCreation(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx)
+	return shared.ValidateDupCreation(ctx, v.Client, &req, gvk.DSCInitialization.Kind, log)
+}
+
+// validateDeletion allows deletion of DSCInitialization only if no DataScienceCluster exists.
+// For unknown kinds, the operation is allowed by default (defensive).
+func (v *Validator) validateDeletion(ctx context.Context, req admission.Request) admission.Response {
+	if req.Kind.Kind != gvk.DSCInitialization.Kind {
```

Comment:
it seems this can be moved (if actually needed) to the `Handle` method, I think this still holds true for the `CREATE` verb

Reply:
Yes, I can remove those and simplify the code if we are not considering further logic in both cases.

Reply:
I'll simplify the validation webhook for datascirncecluster as well.